internal/manifests/collector/service.go

```diff
@@ -84,10 +83,7 @@ func MonitoringService(params manifests.Params) (*corev1.Service, error) {
 		return nil, err
 	}
 
-	_, metricsPort, err := params.OtelCol.Spec.Config.Service.MetricsEndpoint()
-	if err != nil {
-		return nil, err
-	}
+	_, metricsPort := params.OtelCol.Spec.Config.Service.MetricsEndpoint(params.Log)
```

Comment:
Looking at this code again, I'm not wholly sure what the right thing to do here is. I thought the webhook shouldn't reject Collector CRs just because it can't parse the port, but this is the monitoring Service which can be enabled or disabled via a CR attribute. So, to some extent, the user should *want* to get an error here if this won't actually work.

@pavolloffay @jaronoff97 @frzifus what do you think we should do here? The options are

1. Do what we do right now, return an error, and just make sure we handle the case where the host comes from an env variable gracefully, like what this PR originally did.
2. Emit a warning, and simply do not create the monitoring Service if we can't get the port, but otherwise continue with manifest generation.

Reply:
FYI, trying to add IPv6 support to this makes it even more complicated. I gave up on making my patch IPv6 compatible pretty quickly. 😅

Reply:
I'd go with the first option suggested by @swiatekm, but also add a doc for the users clarifying how the monitoring Service configuration is handled.

Reply:
I believe the simplest way forward is to log a warning when we cannot parse the host-port pair in any way to let people know that service creation will be skipped because of this and create the collector.

A slightly better, and yet still simple, approach could be:

- if env vars are detected through the presence of `${}`, skip service creation, warn user about it, and create the Collector anyway.
- if no env vars are detected it means we should 100% be able to parse the host and port, apply defaults, and creator the Collector.

WDYT, @yuriolisa and @swiatekm?

Reply:
Just a small note: the port could also be an env var, which is totally valid Collector configuration and thus we should still accept it.

Reply:
@douglascamata if the port value is an env var reference, then we can't know the value, and we can't create a monitoring Service for it. In that case, I think it's correct for us to return an error. If the user doesn't want that, they can disable the Service creation in the Collector CR.

Reply:
Due to that, I suggested following the first option mentioned above.

Reply:
@swiatekm are you sure Service creation can be disabled by the user in the Collector CR? I couldn't find anything related there. 🤔

Reply:
Yes, it can be controlled via the `observability.metrics.enableMetrics` attribute: https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollectorspecobservabilitymetrics.

Reply:
Ah, that's why I didn't find it: the name of this configuration option is not closely related to what its description says it does.

Thanks for the reference. I'm mentioning this in the README. 👍

Reply:
I mean, it says "enableMetrics" and disabling it in fact prevents the creation of Pod/Service Monitor and Service objects. But the Collector's metrics are still being exported.